facade/src/main/java/org/jboss/pnc/facade/providers/BuildProviderImpl.java

```diff
@@ -236,6 +245,27 @@ private BuildRecord getBuildRecord(int id) {
         }
     }
 
+    @Override
+    public Build getSpecific(Integer id) {
+
+        Build build = super.getSpecific(id);
+
+        if (build != null) {
+
+            return build;
+
+        } else {
+
+            List<BuildTask> runningBuilds = buildCoordinator.getSubmittedBuildTasks();
+
+            return runningBuilds.stream()
+                    .filter(buildTask -> id.equals(buildTask.getId()))
+                    .findAny()
+                    .map(buildMapper::fromBuildTask)
+                    .orElse(null);
+        }
```

Comment:
Can it happen that build is both running and in DB? I would think that trying to find a build among running builds would be faster than in DB, so I would swap the order.

Reply:
It cannot happen that the running build is also in the DB. The Build record in DB is created when the build completes.

Reply:
I've changed the order anyway. But yeah as Matej said, it can only be present in the running or in the DB, but not both.